src/components/DropDownMenu.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useMemo, useRef, useState } from 'react';
+import { Person } from '../types/Person';
+import cn from 'classnames';
+import debounce from 'lodash.debounce';
+
+type Props = {
+  persons: Person[];
+  onSelected: (person: Person | null) => void;
+  delay?: 300;
+};
+
+export const DropDownMenu: React.FC<Props> = ({
+  persons,
+  onSelected,
+  delay = 300,
+}) => {
+  const [focused, setFocused] = useState(false);
+  const [query, setQuery] = useState('');
+  const [delayQuery, setDelayQuery] = useState('');
+  const inputRef = useRef<HTMLInputElement | null>(null);
+  const delayedQuery = debounce(setDelayQuery, delay);
+
+  const handlePersonChange = (selectedPerson: Person) => {
+    setQuery(selectedPerson.name);
+    onSelected(selectedPerson);
+    setFocused(false);
+  };
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
let's also implement this requirement:

![image](https://github.com/mate-academy/react_autocomplete/assets/86923089/21b515db-793d-41ac-917d-82e8cc45742d)